x-pack/plugins/security_solution/public/timelines/components/timeline/tabs/query/index.tsx

```diff
@@ -256,79 +217,27 @@ export const QueryTabContentComponent: React.FC<Props> = ({
     );
   }, [loadingSourcerer, timelineId, isQueryLoading, dispatch]);
 
-  const leadingControlColumns = useMemo(
-    () =>
-      getDefaultControlColumn(ACTION_BUTTON_COUNT).map((x) => ({
-        ...x,
-        headerCellRender: HeaderActions,
-      })),
-    [ACTION_BUTTON_COUNT]
-  );
-
   // NOTE: The timeline is blank after browser FORWARD navigation (after using back button to navigate to
   // the previous page from the timeline), yet we still see total count. This is because the timeline
   // is not getting refreshed when using browser navigation.
   const showEventsCountBadge = !isBlankTimeline && totalCount >= 0;
 
-  const header = useMemo(
-    () => (
-      <StyledEuiFlyoutHeader data-test-subj={`${activeTab}-tab-flyout-header`} hasBorder={false}>
-        <InPortal node={timelineEventsCountPortalNode}>
-          {showEventsCountBadge ? <EventsCountBadge>{totalCount}</EventsCountBadge> : null}
-        </InPortal>
-        <EuiFlexGroup gutterSize="s" direction="column">
-          {!unifiedComponentsInTimelineEnabled &&
-            timelineFullScreen &&
-            setTimelineFullScreen != null && (
-              <EuiFlexItem>
-                <EuiFlexGroup alignItems="center" gutterSize="s">
-                  <ExitFullScreen
-                    fullScreen={timelineFullScreen}
-                    setFullScreen={setTimelineFullScreen}
-                  />
-                </EuiFlexGroup>
-              </EuiFlexItem>
-            )}
-          <EuiFlexItem data-test-subj="timeline-date-picker-container">
-            <TabHeaderContainer data-test-subj="timelineHeader">
-              <QueryTabHeader
-                filterManager={timelineFilterManager}
-                show={show && activeTab === TimelineTabs.query}
-                showCallOutUnauthorizedMsg={showCallOutUnauthorizedMsg}
-                status={status}
-                timelineId={timelineId}
-              />
-            </TabHeaderContainer>
-          </EuiFlexItem>
-          {/* TODO: This is a temporary solution to hide the KPIs until lens components play nicely with timelines */}
-          {/* https://github.com/elastic/kibana/issues/17156 */}
-          {/* <EuiFlexItem grow={false}> */}
-          {/*   <TimelineKpi timelineId={timelineId} /> */}
-          {/* </EuiFlexItem> */}
-        </EuiFlexGroup>
-      </StyledEuiFlyoutHeader>
-    ),
-    [
-      activeTab,
-      timelineFilterManager,
-      show,
-      showCallOutUnauthorizedMsg,
-      status,
-      timelineId,
-      setTimelineFullScreen,
-      timelineFullScreen,
-      unifiedComponentsInTimelineEnabled,
-      timelineEventsCountPortalNode,
-      showEventsCountBadge,
-      totalCount,
-    ]
-  );
-
   if (unifiedComponentsInTimelineEnabled) {
     return (
       <UnifiedTimelineBody
-        header={header}
-        columns={augumentedColumnHeaders}
+        header={
+          <QueryTabHeader
```

Comment:
Couldn't decide whether I wanted this behind a memoized const and just pass those two values instead of duplicating them. I can go either way...

Reply:
+1 for having it in `useMemo`. Lot of things going on in timeline and that might recreate header everytime. 

Which values are you talking about?